hypershift/hosted-wrapper.py

```diff
@@ -256,59 +268,65 @@ def get_metadata(kubeconfig, my_path, duration, cluster_name, uuid, operation):
-    metadata["status"] = metadata_hosted_info['status']['version']['history'][0]['state']
-    metadata["version"] = metadata_hosted_info['status']['version']['history'][0]['version']
     metadata["operation"] = operation
     metadata["uuid"] = uuid
     return metadata
 
 
-def _watcher(kubeconfig_location, cluster_name_seed, cluster_count, delay, my_uuid, clusters_resume):
+def _watcher(kubeconfig_location, cluster_name_seed, cluster_count, delay, my_uuid, clusters_resume, all_clusters_installed):
     myenv = os.environ.copy()
     myenv["KUBECONFIG"] = kubeconfig_location
     time.sleep(60)
     logging.info('Watcher thread started')
     logging.info('Getting status every %d seconds' % int(delay))
-    my_thread = threading.current_thread()
     # We need to determine somewhere the number of clusters to show
-    cmd = ["oc", "get", "hostedclusters", "-n", "clusters"]
+    cmd = ["oc", "get", "hostedclusters", "-n", "clusters", "-o", "json"]
     # To stop the watcher we expect the run attribute to be not True
-    while getattr(my_thread, "run", True):
+    while True:
         logging.debug(cmd)
         process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=myenv)
         stdout, stderr = process.communicate()
-        current_cluster_count = 0
-        state = {}
-        error = []
-        # Count the various states/status' and report it to logging
-        for line in stdout.splitlines():
-            if cluster_name_seed in line:
-                if len(line.split()) >= 3:
-                    current_cluster_count += 1
-                    state_key = line.split()[3] if 'Completed' in line else line.split()[2]
-                    state[state_key] = state.get(state_key, 0) + 1
-                    # Not sure about the error state key
-                    if state_key == "error":
-                        error.append(line.split()[0])
-
-        logging.info('Requested Clusters for test %s: %d' % (my_uuid, cluster_count))
-        if current_cluster_count != 0:
-            logging.debug(state.items())
-            state_output = "Current clusters state: " + str(current_cluster_count) + " clusters"
-            for i in state.items():
-                state_output += " (" + str(i[0]) + ": " + str(i[1]) + ")"
-            logging.info(state_output)
-            if error:
-                logging.warning('Clusters in error state: %s' % error)
-            clusters_resume['state'] = state
-            clusters_resume['clusters_created'] = current_cluster_count
-        time.sleep(delay)
+        installing_clusters = 0
+        installed_clusters = 0
+        other_clusters = 0
+        other_list = []
+        for cluster in json.loads(stdout)['items']:
+            if cluster_name_seed in cluster['metadata']['name']:
+                status = cluster['status']['version']['history'][0]
```

Comment:
This could technically explode if that key does not exist in the cluster json. similarly with the status[state] stuff below.

Reply:
Added try catch on json load and protect any assignment by previously checking if the key exists